docs/reference-capabilities/default-capabilities.md

```diff
@@ -0,0 +1,233 @@
+
+```pony
+actor UsesStrings
+    let s1: String
+    var s2: String
+
+    new create(s1': String, s2': String) =>
+        s1 = s1'
+        s2 = s2'
+
+    be do_thing(s3: String) =>
+        None
+
+```
+
+However `StringBytes` is defined as `class ref`, so all the `StringBytes`' in the following class definition are equivalent to `StringBytes ref`:
+
+```pony
+class UsesStringBytes
+    let s1: StringBytes
+    var s2: StringBytes
+
+    new create(s1': StringBytes, s2': StringBytes) =>
+        s1 = s1'
+        s2 = s2'
+
+    fun do_thing(s3: StringBytes) =>
+        None
+```
+
+Notably, however, the default capability on a class **does not** influence the constructors for that class.
+
+## Constructor capabilities
+
+Let's start with the easy defaults first: A primitive constructor _always_ returns an object with `val` capabilities, and an actor constructor _always_ returns an object with `tag` capabilities. In fact, it is forbidden to put a capability after `new` when defining a primitive or actor.
+
+Classes are a little more flexible. The capability returned by a `new` constructor default to a `ref` if you are constructing a `class`. This is true regardless of what the default capability for the class is, but you have the option to specify a different capability by explicitly typing the constructor as e.g `new var create` or `new iso create_iso`.
+
+Consider the following class:
+
+```pony
+class val MyClass
+    let foo: String
+
+    new create_ref(foo': String) =>
+        foo = foo'
+
+    new val create_val(foo': String) =>
+        foo = foo'
+```
+
+This class has two constructors. The first one, `create_ref` returns a `MyClass` with `ref` capabilities because `ref` is always default for class constructors. The second one `create_val` explicitly returns a `MyClass` with `val` capabilities.
+
+_Tip_: Instead of using the default `ref` type for constructors, consider using `iso`. Recall that constructors, like `consume` return their values as an ephemeral type. That means that a constructor typed as `new iso` actually returns a `iso^`. And `iso^` has the terrific property of being convertible to any other type! Consider:
+
+```pony
+class MyIsoClass
+    let foo: String
+
+    new iso create(foo': String) =>
+        foo = foo'
+```
+
+This class can be constructed as a ref or val or whatever else:
+
+```pony
+a_ref: MyIsoClass ref = MyIsoClass.create()
+a_val: MyIsoClass val = MyIsoClass.create()
+```
+
+In fact, this is what happens if you do not specify a constructor at all. The hidden constructor for this class also returns an `iso^`:
+
+```pony
+class AllDefaults
+    var foo: String = "hello"
+```
+
+You can construct instances of this class with a specific capability by specifying the type of the result:
```

Comment:
Does not including a `new create()` here cause it to be `iso` by default? If so, that is news to me and may be why I was confused just above.

Reply:
`new create()` is not `new iso create()` it is `new ref create()`.

Reply:
oops, so it is. I changed this to:

```
let a = AllDefaults // a is ref
let b: AllDefaults iso = AllDefaults // b is iso
let c: AllDefaults val = AllDefaults // c is val
```

However, I'm leaving the discussion open, as I realized the entire constructors section is overlooking the fact that constructors return ephemeral types.

Reply:
Ok, I'm confused. Consider this code:

```pony
let a = AllDefaults
let a2: AllDefaults val = a
```

This is obviously a compile-time error. And here it is:

```
Error:
/Users/dustyphillips/Desktop/Code/pony-experiments/referee/referee.pony:66:33: right side must be a subtype of left side
        let a2: AllDefaults val = a
                                ^
    Info:
    /Users/dustyphillips/Desktop/Code/pony-experiments/referee/referee.pony:157:1: AllDefaults iso is not a subtype of AllDefaults val^: iso is not a subcap of val^
    class AllDefaults
    ^
    /Users/dustyphillips/Desktop/Code/pony-experiments/referee/referee.pony:157:1: this would be possible if the subcap were more ephemeral. Perhaps you meant to consume a variable here
    class AllDefaults
    ^
```

The reason I thought `a` in this case has `iso` capabilities is that phrase `iso is not a subcap of val^`.

Reply:
Sorry, what are you confused by?